File: frontend/app/types/arquivo.ts
import type { ModoRetencaoArquivo } from '~/types/politicaRetencaoArquivo'
export type { ModoRetencaoArquivo }

export type TipoArquivo =
  | 'extrato_bancario'
  | 'planilha_interna'

export type StatusArquivo =
  | 'enviado'
  | 'validado'
  | 'invalido'
  | 'processado'
  | 'erro'

export interface ArquivoEnviado {
  id: string
  empresa_id: string
  conciliacao_id: string
  nome_original: string
  tipo_arquivo: TipoArquivo | string
  status: StatusArquivo | string
  tamanho_bytes: number
  // Campos de retenção (nullable em arquivos anteriores à migration 009)
  modo_retencao?: ModoRetencaoArquivo | null
  arquivo_persistido?: boolean
  expira_em?: string | null
  excluido_em?: string | null
  hash_arquivo?: string | null
  metadados?: Record<string, unknown>
  arquivo_disponivel?: boolean
  permitir_download_original?: boolean
  permitir_reprocessamento_sem_reenvio?: boolean
  criado_em: string
}

export const tipoArquivoLabels: Record<string, string> = {
  extrato_bancario: 'Extrato Bancário',
  planilha_interna: 'Fluxo de Caixa',
}

export const statusArquivoLabels: Record<string, string> = {
  enviado:   'Enviado',
  validado:  'Validado',
  invalido:  'Inválido',
  processado:'Processado',
  erro:      'Erro',
}

File: backend/app/schemas/arquivo.py
import uuid
from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel

TipoArquivo = Literal[
    "extrato_bancario", "relatorio_vendas", "relatorio_recebiveis",
    "planilha_interna", "taxas_adquirente", "outro",
]

StatusArquivo = Literal["enviado", "lido", "invalido", "processado", "erro"]

ModoRetencao = Literal["somente_memoria", "temporario", "persistente"]


class ArquivoResponse(BaseModel):
    id: uuid.UUID
    empresa_id: uuid.UUID
    conciliacao_id: uuid.UUID
    nome_original: str
    tipo_arquivo: str
    status: str
    tamanho_bytes: int
    # Campos de retenção — nullable para arquivos anteriores à migration 009
    modo_retencao: ModoRetencao | None = None
    arquivo_persistido: bool = True
    expira_em: datetime | None = None
    excluido_em: datetime | None = None
    hash_arquivo: str | None = None
    metadados: dict[str, Any] = {}
    arquivo_disponivel: bool = True
    permitir_download_original: bool = True
    permitir_reprocessamento_sem_reenvio: bool = True
    criado_em: datetime

    model_config = {"from_attributes": True}
